### ISSUE TITLE
The program has no problem on a single GPU, and errors are reported on multiple GPUs.

### ISSUE THREAD

**Issue by zjr35897** [microsoft/MASS#61]
python3.5
pytorch-gpu 1.2
cuda 10.0
when I run the this program,for example:

> python -m torch.distributed.launch --nproc_per_node=$NGPU train.py \   
--exp_name unsupMT_enfr \                               
--data_path ./data/processed/en-fr/   \                 
--lgs 'en-fr'             \                             
--bt_steps 'en-fr-en,fr-en-fr'     \                 
 --encoder_only false                \                  
 --emb_dim 1024                       \                 
 --n_layers 6                              \            
 --n_heads 8                             \               
--dropout 0.1                          \                
--attention_dropout 0.1         \                       
--gelu_activation true              \                  
 --tokens_per_batch 2000         \                       
--batch_size 8                          \             
--bptt 256                                 \           
 --optimizer adam_inverse_sqrt,beta1=0.9,beta2=0.98,lr=0.0001   --epoch_size 200000        \                            
--max_epoch 30                     \                    
--eval_bleu true                       \                
--reload_model "$MODEL,$MODEL" \

there are some problems:
![image](https://user-images.githubusercontent.com/18071348/64757371-8f5a0c80-d564-11e9-8d66-9de1f6b75b7f.png)
But it works well when i use single GPU.
what's the problem?

**Comment by StillKeepTry** [comment]
Maybe add `export NGPU=8` in your system variable.

**Comment by zjr35897** [comment]
I add NGPU=2, but it didn't work

**Comment by StillKeepTry** [comment]
Can you switch the version of pytorch into torch 1.0 to have a try?

**Comment by huangxianliang** [comment]
i have an other problem with pretraining: python3.5 and pytorch-gpu 1.0,command is export NGPU=8; $PYTHON -m torch.distributed.launch --nproc_per_node=$NGPU train.py,i had not set local_rank(default is -1),
![image](https://user-images.githubusercontent.com/17400821/65825865-a0b64f00-e2ae-11e9-872d-6adb25e55f92.png)

**Comment by StillKeepTry** [comment]
@huangxianliang Do you solve your problem. It seems like you are using Slurm job ?
